SecuritySystemUWP/SecuritySystemUWP/Camera/Webcam.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Devices.Gpio;
+using Windows.Storage;
+using System.Collections.Generic;
+using Microsoft.Maker.Devices.Media.UsbCamera;
+using Microsoft.Maker.Devices.Gpio.PirSensor;
+
+namespace SecuritySystemUWP
+{
+    public class Webcam : ICamera
+    {
+        private UsbCamera webcam;
+        private PirSensor pirSensor;
+        private bool isCapturing;
```

Comment:
this variable is being initialized, modified, and and read in different functions, some of which can run asyncronously. To guarantee proper visibility between threads, either make this variable `volatile` (not recommended, but sufficient) or use proper object locking mechanisms when reading/writing